File: cs50/pset6/readability/readability.py
# Packages
from cs50 import get_string
import re

# Define main function
def main():

    # 1. Ask for text
    text = get_string('Text: ')

    # 2. Count letters, words, and sentences
    letters = 0
    words = 1
    sentences = 0

    for i in text:

        if re.search('[a-zA-Z]', i):
            letters += 1

        elif re.search(' ', i):
            words += 1

        elif re.search('[.!?]', i):
            sentences += 1

    # 3. Compute index
    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8

    # 4. Output corresponding text
    if index < 1:
        print("Before Grade 1")

    if index >= 1 and index < 16:
        print("Grade", round(index))

    if index >= 16:
        print("Grade 16+")

main()

File: cs50/pset2/readability/readability.c
// Packages
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>

// Main function
int main(void)
{
    // 1. Input text
    string text = get_string("Text: ");
    
    // 2. Compute the number of letters, words, and sentences
    float letters = 0;
    float words = 1;
    float sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
        if (text[i] == ' ')
        {
            words++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    //printf("%f letter(s)\n", letters);
    //printf("%f words(s)\n", words);
    //printf("%f sentences(s)\n", sentences);

    // 3. Compute Coleman-Liau index
    float L = letters / words * 100;
    float S = sentences / words * 100;
    float index = round(0.0588 * L - 0.296 * S - 15.8);
    //printf("L: %f\n", L);
    //printf("S: %f\n", S);
    //printf("Index: %f\n", index);
    
    // 4. Print out grade result
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    if (index >= 1 && index < 16)
    {
        printf("Grade %i\n", (int) index);
    }
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
}